src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java

```diff
@@ -1100,4 +1100,50 @@ public String toString() {
       public static final String PACKET_PER_SEC_VALUE = "packets/s";
     }
   }
+
+  public enum ShardStateDimension implements MetricDimension {
+    SHARD_ID(CommonDimension.SHARD_ID.toString()),
+    INDEX_NAME(CommonDimension.INDEX_NAME.toString()),
+    SHARD_TYPE(Constants.SHARD_TYPE),
+    NODE_NAME(Constants.NODE_NAME);
```

Comment:
Because the writer only emits the node local metrics, the node name is implicit. This is the reason we don't have it [here](https://opendistro.github.io/for-elasticsearch-docs/docs/pa/reference/#dimensions-reference). When we gather the metrics from all the nodes, it can be added. Because it is the same for all the metrics, we don't want to store it as it will save us bytes on the disk (as not having the SQLIte column) and bytes on the heap when we read the table into memory.